sdk/log/record.go

```diff
@@ -131,6 +150,26 @@ func (r *Record) WalkAttributes(f func(log.KeyValue) bool) {
 
 // AddAttributes adds attributes to the log record.
 func (r *Record) AddAttributes(attrs ...log.KeyValue) {
+	if r.attributeCountLimit > 0 {
+		if r.AttributesLen()+len(attrs) > r.attributeCountLimit {
+			r.compactAttr()
+			// TODO: apply truncation to string and []string values.
+			var dropped int
+			attrs, dropped = deduplicate(attrs)
```

Comment:
should we be deduplicating against the existing attributes as well?

Reply:
Updated. PTAL.